### PR TITLE
Improve avatar components

### PR DESCRIPTION
**Pull request by liviu-timar**
### 🎯 Goal

Improve avatar components:
- reviewed avatar components and possible usages
- auto font size adjustment for initials avatars
- clean up parameter lists
- change visibility for similar components
- add/improve KDocs.

### 🛠 Implementation details

- Detected when text is overflowing in `Text` composable in `InitialsAvatar` and gradually decreased font size if needed (decrease step can be added as parameter in the future). Parameter with type `TextStyle.fontSize` is used as starting size.
- Changed `Avatar` visibility to `internal`, as we also have `UserAvatar` that basically does the same thing.
- Improved naming and order of parameters.
- Improved KDocs in several components as we had references to Chat components (probably copy-pasted from Chat).

### 🎨 UI Changes

| Before | After |
| --- | --- |
| Passed-in (or default) font size is too big, so the text overflows | The font size is decreased automatically until no more overflow |
| before-1 | Screenshot_20240419_135722 |
| before-2 | Screenshot_20240419_135747 |
| before-3 | Screenshot_20240419_135753 |
| Screenshot 2024-04-19 at 14 49 31 | Screenshot 2024-04-19 at 14 47 53 |

### 🧪 Testing

Check both image & initials avatars with different sizes. Can be done throughout the demo app and the audio call sample app.

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [x] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)
- [ ] Tutorial starter kit updated
- [ ] Examples/guides starter kits updated (`stream-video-examples`)

### ☑️Reviewer Checklist
- [ ] XML sample runs & works
- [ ] Compose sample runs & works
- [ ] Tutorial starter kit
- [ ] Example starter kits work
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

**Comment by liviu-timar** [comment]
Should we use `@Deprecated` for changes (parameters deleted or name changed, component marked as internal etc)?

**Comment by liviu-timar** [comment]
I moved the public `ParticipantAvatars` out of the `participants/internal` package. Extra handling needed?

File: stream-video-android-ui-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/avatar/UserAvatar.kt [hunk 1]
```diff
@@ -34,46 +34,50 @@ import androidx.compose.ui.unit.IntSize
...
 public fun UserAvatar(
-    userName: String?,
-    userImage: String?,
     modifier: Modifier = Modifier,
+    userImage: String? = null,
+    userName: String? = null,
     shape: Shape = VideoTheme.shapes.circle,
-    contentScale: ContentScale = ContentScale.Crop,
-    contentDescription: String? = null,
-    textSize: StyleSize = StyleSize.XL,
```
Comment (liviu-timar) [inline stream-video-android-ui-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/avatar/UserAvatar.kt:72]: Should we use `@Deprecated` for modified composables (parameters deleted or names changed, function marked as `internal` etc)? I’ve made changes in these public components: `Avatar`, `UserAvatar`, `UserAvatarBackground`, `ParticipantAvatars`.

File: stream-video-android-ui-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/avatar/Avatar.kt [hunk 2]
```diff
@@ -44,45 +44,40 @@ import com.skydoves.landscapist.coil.CoilImage
...
+ * @param imageRequestSize The image size to be requested.
+ * @param loadingPlaceholder Placeholder image to be displayed while loading the remote image.
+ * @param previewModePlaceholder Placeholder image to be displayed in Compose previews (IDE).
+ * @param textStyle The text style of the [fallbackText] text. The `fontSize`, `fontFamily` and `fontWeight` properties are used.
+ * If the font size is too large, it will be gradually decreased automatically.
+ * @param textOffset Offset to be applied to the initials text.
+ * @param onClick Handler to be called when the user clicks on the avatar.
  */
 @Composable
-public fun Avatar(
```
Comment (liviu-timar) [inline stream-video-android-ui-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/avatar/Avatar.kt:68]: Should we make the `Avatar` component `internal`? We already have `UserAvatar` as `public` and it basically does the same thing. Impact? See [this](https://github.com/GetStream/stream-video-android/pull/1072/commits/ac991ab80d9415f44421cd8da83461a5a0ca474f) commit.

File: stream-video-android-ui-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/participants/ParticipantAvatars.kt [hunk 3]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package io.getstream.video.android.compose.ui.components.participants.internal
```
Comment (liviu-timar) [inline stream-video-android-ui-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/participants/ParticipantAvatars.kt:17]: I moved the public `ParticipantAvatars` component out of the `participants/internal` package. Extra handling needed?

**Review by aleksandar-apostolov — APPROVED** [review]

File: stream-video-android-ui-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/avatar/Avatar.kt (hunk 2 above)
Comment (aleksandar-apostolov) [reply stream-video-android-ui-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/avatar/Avatar.kt:68]: We'll revise this at a later stage.

**Pull request merged**